src/protocols/bdx/BdxMessages.cpp

```diff
@@ -0,0 +1,456 @@
+
+    return (Version == another.Version && TransferCtlFlags.Raw() == another.TransferCtlFlags.Raw() &&
+            MaxBlockSize == another.MaxBlockSize && metadataMatches);
+}
+
+void ReceiveAccept::Pack(BufBound & aBuffer) const
+{
+    uint8_t transferCtl = 0;
+    bool widerange      = (StartOffset > std::numeric_limits<uint32_t>::max()) || (Length > std::numeric_limits<uint32_t>::max());
+
+    transferCtl |= Version & kVersionMask;
+    transferCtl = transferCtl | TransferCtlFlags.Raw();
+
+    BitFlags<uint8_t, RangeControlFlags> rangeCtlFlags;
+    rangeCtlFlags.Set(kDefLen, Length > 0);
+    rangeCtlFlags.Set(kStartOffset, StartOffset > 0);
+    rangeCtlFlags.Set(kWiderange, widerange);
+
+    aBuffer.Put(transferCtl);
+    aBuffer.Put(rangeCtlFlags.Raw());
+    aBuffer.PutLE16(MaxBlockSize);
+
+    if (StartOffset > 0)
+    {
+        if (widerange)
+        {
+            aBuffer.PutLE64(StartOffset);
+        }
+        else
+        {
+            aBuffer.PutLE32(static_cast<uint32_t>(StartOffset));
+        }
+    }
+
+    if (Length > 0)
+    {
+        if (widerange)
+        {
+            aBuffer.PutLE64(Length);
+        }
+        else
+        {
+            aBuffer.PutLE32(static_cast<uint32_t>(Length));
+        }
+    }
+
+    if (Metadata != nullptr)
+    {
+        aBuffer.Put(Metadata, static_cast<size_t>(MetadataLength));
+    }
+}
+
+CHIP_ERROR ReceiveAccept::Parse(const System::PacketBuffer & aBuffer)
+{
+    CHIP_ERROR err          = CHIP_NO_ERROR;
+    uint8_t transferCtl     = 0;
+    uint8_t rangeCtl        = 0;
+    uint32_t tmpUint32Value = 0; // Used for reading non-wide length and offset fields
+    uint8_t * bufStart      = aBuffer.Start();
+    Reader bufReader(bufStart, aBuffer.DataLength());
+    BitFlags<uint8_t, RangeControlFlags> rangeCtlFlags;
+
+    SuccessOrExit(bufReader.Read8(&transferCtl).Read8(&rangeCtl).Read16(&MaxBlockSize).StatusCode());
+
+    Version = transferCtl & kVersionMask;
+
+    // Only one of these values should be set. It is up to the caller to verify this.
+    TransferCtlFlags.SetRaw(static_cast<uint8_t>(transferCtl & ~kVersionMask));
+
+    rangeCtlFlags.SetRaw(rangeCtl);
+
+    if (rangeCtlFlags.Has(kStartOffset))
+    {
+        if (rangeCtlFlags.Has(kWiderange))
+        {
+            SuccessOrExit(bufReader.Read64(&StartOffset).StatusCode());
+        }
+        else
+        {
+            SuccessOrExit(bufReader.Read32(&tmpUint32Value).StatusCode());
+            StartOffset = tmpUint32Value;
+        }
+    }
+
+    if (rangeCtlFlags.Has(kDefLen))
+    {
+        if (rangeCtlFlags.Has(kWiderange))
+        {
+            SuccessOrExit(bufReader.Read64(&Length).StatusCode());
+        }
+        else
+        {
+            SuccessOrExit(bufReader.Read32(&tmpUint32Value).StatusCode());
+            Length = tmpUint32Value;
+        }
+    }
+
+    // Rest of message is metadata (could be empty)
+    if (bufReader.Remaining() > 0)
```

Comment:
Same comment here re: making sure all fields have been set (including `Metadata = nullptr` and `MetadataLength = 0` by the time success case is hit.

Reply:
i'll make sure Parse() fills out all fields